src/Projection/GapDetection.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/**
+ * This file is part of prooph/pdo-event-store.
+ * (c) 2016-2020 Alexander Miertsch <kontakt@codeliner.ws>
+ * (c) 2016-2020 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStore\Pdo\Projection;
+
+use Prooph\Common\Messaging\Message;
+
+/**
+ * Class GapDetection
+ *
+ * In high load scenarios it is possible that projections skip events due to newer events becoming visible before old ones.
+ * For details you can take a look at this issue: https://github.com/prooph/pdo-event-store/issues/189
+ *
+ * The GapDetection class helps projectors to detect gaps while processing events and coordinates retries to fill gaps.
+ *
+ * @package Prooph\EventStore\Pdo\Projection
+ */
+final class GapDetection
```

Comment:
We need an interface so we can test against mocks.

Reply:
I don't think the class needs to be mocked. It is basically a wrapper around some config options and two helper methods. See current test case. Works without mocking.